tests/testImageMapReduce.py

```diff
@@ -377,6 +377,35 @@ def testGridCentroidsWrongLength(self):
         with self.assertRaises(ValueError):
             task.run(self.exposure)
 
+    def testMasks(self):
+        """Test the mask for an exposure produced by a sample grid task
+        where we provide a set of `gridCentroids` and thus should have
+        many invalid pixels.
+        """
+        config = AddAmountImageMapReduceConfig()
+        config.gridStepX = config.gridStepY = 8.
+        config.gridCentroidsX = [i for i in np.linspace(0, 128, 50)]
+        config.gridCentroidsY = config.gridCentroidsX
+        config.reducerSubtask.reduceOperation = 'average'
+        task = ImageMapReduceTask(config)
+        config.mapperSubtask.addAmount = 5.
+        newExp = task.run(self.exposure).exposure
+        newMI = newExp.getMaskedImage()
+        newArr = newMI.getImage().getArray()
+        mi = self.exposure.getMaskedImage()
+        isnan = np.isnan(newArr)
+        self.assertGreater(np.sum(isnan), 1000)
+
+        mi = self.exposure.getMaskedImage().getImage().getArray()
+        self.assertFloatsAlmostEqual(mi[~isnan], newArr[~isnan] - 5.)
+
+        mask = newMI.getMask()  # Now check the mask
+        self.assertGreater(mask.getMaskPlane('INVALID_MAPREDUCE'), 0)
+        maskBit = mask.getPlaneBitMask('INVALID_MAPREDUCE')
+        nMasked = np.sum((mask.getArray()[:, :] & maskBit) != 0)
```

Comment:
Im pretty sure here the operation you want is numpy.bitwise_and, you can avoid the [:,:] in that case.